packages/vscode-extension/src/extension.ts

```diff
@@ -46,13 +74,93 @@ export function activate(context: ExtensionContext) {
     clientOptions,
   );
 
+  // Initialize singletons
+  LangugageClientManager.instance = new LangugageClientManager(client);
+  DatabaseDriverManager.instance = new DatabaseDriverManager();
+  ConnectionRepository.instance = new ConnectionRepository(
+    context.globalState,
+    context.secrets,
+  );
+
+  await ConnectionRepository.instance.resetConnections();
+
+  // Register our tree view provider to render connections in the sidebar
+  const connectionTreeDataProvider = new ConnectionTreeDataProvider();
+
+  // Register commands
+  context.subscriptions.push(
```

Comment:
I think I may create a commands object that handles these and tidies up this activation function